### PR TITLE
feat: new title bar design

### LINKED ISSUE #377: Simplification of the Design: Too many Title Bars?
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

**Is your feature request related to a problem? Please describe.**
- Currently, the desktop chat application includes a standard title bar at the top of the window, which displays the application name and provides window management options (e.g., minimize, maximize, close buttons).
- Also, there is the web-like main bar with a big "Nextcloud Talk preview", search, notification and self-menu icons.
- Finally, there is the conversation-specific bar (conversation settings, start call button...).

I don't know exactly why, but I find the combinaison of three bars being weird and overwhelming...
Also, there is too much corporate text in my opinion.

Maybe we could have an option to hide or make this three bars more integrated to provide a cleaner and more streamlined user experience. For enhanced aesthetic and modern design.

**Describe the solution you'd like**
An option within the application's settings or preferences menu that allows users to toggle the visibility of the title bars on or off. This way, users who prefer the traditional window layout with the title bar can keep it, while those who prefer a more streamlined interface can choose to hide it.
I don't see a perfect solution here, but maybe some ideas will emerge from the discussion.

**Describe alternatives you've considered**

Current Talk:
image

Some options from Messenger, WhatsApp, Telegram and Signal:
Corporate titles or icons are very small, and they only have two bars. 
Messenger:
image

WhatsApp:
image

Telegram:
image

Signal:
image

**Additional context**
It's an open suggestion!

### PR DESCRIPTION
**Pull request by ShGKme**
## Resolves

- Fix #377

### 🏚️ Before - everywhere

image

### 🏡 After on Windows 🪟

image

### 🏡 After on Mac 🍎

image

### 🏡 After on Linux 🐧

image

### With viewer

https://github.com/nextcloud/talk-desktop/assets/25978914/30e24a44-7257-4f3e-bd3b-d69e0b591465

### Login Window

Before | After
---|---
image | image
image | image

### Welcome Window on Mac

Before | After
---|---
image | image

### CODE REVIEW

**Review by Antreesy — APPROVED** [review]
Tested on Linux and Mac.

Some Mac window headers seem to have incorrect width (not sure if related):
image

Apart from that, working and code-wise good.

File: src/talk/renderer/DesktopHeader.vue [hunk 1]
```diff
@@ -20,74 +20,55 @@
   -->
 
 <template>
-	<header class="header">
-		<!-- Nextcloud Apps use #header selector to get its height -->
-		<!-- This is invisible stub with the same height -->
-		<div id="header" class="header-stub" />
-
-		<a class="header__root-link" @click.prevent="pushToRoot">
```
Comment (nickvergessen) [inline src/talk/renderer/DesktopHeader.vue:28]: No more option to go back to root with the mouse?

Comment (ShGKme) [reply src/talk/renderer/DesktopHeader.vue:28]: Now you can do it with <kbd>Esc</kbd>.

It wasn't obvious anyway, I'm surprised someone else except me even used this feature :D

**Comment by ShGKme** [comment]
> Some Mac window headers seem to have incorrect width (not sure if related):

What do you mean?

File: src/talk/renderer/DesktopHeader.vue (hunk 1 above)
Comment (nickvergessen) [reply src/talk/renderer/DesktopHeader.vue:28]: The click is quite common and switching from mouse to keyboard while working one-handed is pretty troublesome.
Can we make the "Nextcloud Talk Preview" navigate to root then when it's displayed anyway?

Comment (ShGKme) [reply src/talk/renderer/DesktopHeader.vue:28]: I can, but there is no title on Mac, because it is shown as a separated top bar natively.

Comment (marcoambrosini) [reply src/talk/renderer/DesktopHeader.vue:28]: What's the purpose of navigating to that page @nickvergessen ? 
The sole purpose of that screen is to give some orientation to users when they first open the app, I think it's totally fine if we have no shortcuts to it. 
Are there other plans for that root page that I'm not aware of?

**Review by marcoambrosini — COMMENTED** [review]
Very nice :) 
From a design perspective the only thing I would change is adding slightly more padding to the buttons on the left, similar to this window: 

Screenshot 2024-04-17 at 14 14 03

File: src/talk/renderer/DesktopHeader.vue (hunk 1 above)
Comment (nickvergessen) [reply src/talk/renderer/DesktopHeader.vue:28]: > What's the purpose of navigating to that page

Being immediately inactive in all conversations to not appear as online and to receive notifications

Comment (marcoambrosini) [reply src/talk/renderer/DesktopHeader.vue:28]: > Being immediately inactive in all conversations to not appear as online and to receive notifications

I'd say closing the app covers this use case :)

**Comment by marcoambrosini** [comment]
Yes @ShGKme

**Comment by ShGKme** [comment]
Changes:
- Made title clickable again to go to the root on Windows and Linux (Also works by <kbd>ESC<kbd> everywhere)
- Moved traffic light button on Mac to make them in line with the SearchBox

**Pull request merged**